app/src/main/java/com/into/websoso/ui/termsAgreement/TermsAgreementDialogBottomSheet.kt

```diff
@@ -0,0 +1,160 @@
+package com.into.websoso.ui.termsAgreement
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import androidx.core.os.bundleOf
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.into.websoso.R
+import com.into.websoso.R.drawable.bg_novel_rating_date_primary_100_radius_12dp
+import com.into.websoso.R.drawable.bg_profile_edit_gray_70_radius_12dp
+import com.into.websoso.R.drawable.ic_terms_agreement_selected
+import com.into.websoso.R.drawable.ic_terms_agreement_unselected
+import com.into.websoso.R.string.string_terms_agreement_complete
+import com.into.websoso.R.string.string_terms_agreement_next
+import com.into.websoso.core.common.ui.base.BaseBottomSheetDialog
+import com.into.websoso.databinding.DialogTermsAgreementBinding
+import com.into.websoso.ui.termsAgreement.model.AgreementType
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class TermsAgreementDialogBottomSheet :
+    BaseBottomSheetDialog<DialogTermsAgreementBinding>(R.layout.dialog_terms_agreement) {
+
+    private val termsAgreementViewModel: TermsAgreementViewModel by viewModels()
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setupBottomSheet()
+        onRequireTermsAgreementClick()
+        onTermsAgreementToggleClick()
+        onTermsAgreementCompleteButtonClick()
+        setupViewModel()
+        updateCompleteButtonText()
+    }
+
+    private fun setupBottomSheet() {
+        (dialog as BottomSheetDialog).apply {
+            behavior.state = BottomSheetBehavior.STATE_EXPANDED
+            behavior.isDraggable = false
+            behavior.isHideable = false
+            setCancelable(false)
+        }
+    }
+
+    private fun onRequireTermsAgreementClick() {
```

Comment:
```suggestion
    private fun onRequiredTermsAgreementClick() {
```
c: 처음에 require() 함수를 쓴줄 오해했어요...! 분사형으로 두시는건 어떠세요??

Reply:
으앗 이거 제가 깜빡하고 수정을 안했네염